app/controllers/users.js

```diff
@@ -155,34 +157,74 @@ exports.addDonation = function(req, res) {
 /**
  *  Show profile
  */
-exports.show = function(req, res) {
-  var user = req.profile;
+exports.show = function (req, res) {
+  const user = req.profile;
 
   res.render('users/show', {
     title: user.name,
-    user: user
+    user
   });
 };
 
 /**
  * Send User
  */
-exports.me = function(req, res) {
+exports.me = function (req, res) {
   res.jsonp(req.user || null);
 };
 
 /**
  * Find user by id
  */
-exports.user = function(req, res, next, id) {
+exports.user = function (req, res, next, id) {
   User
     .findOne({
       _id: id
     })
-    .exec(function(err, user) {
+    .exec((err, user) => {
       if (err) return next(err);
-      if (!user) return next(new Error('Failed to load User ' + id));
+      if (!user) return next(new Error(`Failed to load User ${  id}`));
       req.profile = user;
       next();
     });
-};
\ No newline at end of file
+};
+
+/**
+  * @param {any} req
+  * @param {any} res
+  * @return {Object}
+*/
+// Method to Login User
+exports.login = function (req, res) {
+  // Destructure from user
+  const { email, password } = req.body;
+  // Find email
+  User.findOne({ email }).exec((err, user) => {
+    if (err) {
+      return res.status(500).json({
+        error: 'Internal Server Error'
+      });
+    }
+    // If no user found
+    if (!user) {
+      return res.status(400).json({
+        error: 'User Not Found'
+      });
+    }
+    // Compare password from user to database
+    if (bcrypt.compareSync(password, user.hashed_password)) {
+      const userData = {
+        id: user.id
+      };
+      // Create token
+      const token = jwt.sign(userData, 'secretkey', { expiresIn: '5h' });
+      return res.status(200).json({
+        token,
+        message: 'Successfully SignIn',
+      });
+    }
+    return res.status(400).json({
```

Comment:
Based on what you wrote, this status code is incorrect, please change to the correct status code and should be an else block.